From beb5d60922b914b095b75b1ce30196062e574d87 Mon Sep 17 00:00:00 2001
From: Yves-Alexis Perez <corsac@debian.org>
Date: Tue, 12 Jan 2016 17:14:33 -0800
Subject: disable roaming in ssh client

SSH roaming implementation in openssh client is vulnerable to an
information leak (CVE-2016-0777) and heap-based buffer overflow
(CVE-2016-0778). The information leak is somehow attacker-controller,
and it is for example possible to extract the ssh client private keys.

Patch-Name: disable-roaming.patch
---
 readconf.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/readconf.c b/readconf.c
index 29338b6..337d914 100644
--- a/readconf.c
+++ b/readconf.c
@@ -1640,7 +1640,7 @@ initialize_options(Options * options)
 	options->tun_remote = -1;
 	options->local_command = NULL;
 	options->permit_local_command = -1;
-	options->use_roaming = -1;
+	options->use_roaming = 0;
 	options->visual_host_key = -1;
 	options->ip_qos_interactive = -1;
 	options->ip_qos_bulk = -1;
@@ -1822,8 +1822,7 @@ fill_default_options(Options * options)
 		options->tun_remote = SSH_TUNID_ANY;
 	if (options->permit_local_command == -1)
 		options->permit_local_command = 0;
-	if (options->use_roaming == -1)
-		options->use_roaming = 1;
+	options->use_roaming = 0;
 	if (options->visual_host_key == -1)
 		options->visual_host_key = 0;
 	if (options->ip_qos_interactive == -1)
